File: smart-parking-app/parking-availability-service/src/main/java/com/adb/model/Parking.java
package com.adb.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;

import java.io.Serializable;



@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Parking implements Serializable {
   @Id
    private Long id ;
    private  String name;
    private String description ;
    private Boolean availabilityStatus;//construction


}

File: smart_parking_front_end/lib/models/Parking.dart
class Parking {
  final int id;
  final String name;
  final String description;
  final bool availabilityStatus;

  Parking({
    required this.id,
    required this.name,
    required this.description,
    required this.availabilityStatus,
  });

  factory Parking.fromJson(Map<String, dynamic> json) {
    return Parking(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      availabilityStatus: json['availabilityStatus'],
    );
  }
}
